src/MediatR/MicrosoftExtensionsDI/MediatrServiceConfiguration.cs

```diff
@@ -67,8 +67,28 @@ public class MediatRServiceConfiguration
     /// <summary>
     /// Automatically register processors during assembly scanning
     /// </summary>
-    public bool AutoRegisterRequestProcessors { get; set; }
-
+    public bool AutoRegisterRequestProcessors { get; set; }
+
+    /// <summary>
+    /// Configure the maximum number of type parameters that a generic request handler can have. To Disable this constraint, set the value to 0.
+    /// </summary>
+    public int MaxGenericTypeParameters { get; set; } = 10;
+
+    /// <summary>
+    /// Configure the maximum number of types that can close a generic request type parameter constraint.  To Disable this constraint, set the value to 0.
+    /// </summary>
+    public int MaxTypesClosing { get; set; } = 100;
+
+    /// <summary>
+    /// Configure the Maximum Amount of Generic RequestHandler Types MediatR will try to register.  To Disable this constraint, set the value to 0.
+    /// </summary>
+    public int MaxGenericTypeRegistrations { get; set; } = 125000;
+
+    /// <summary>
+    /// Configure the Timeout in Milliseconds that the GenericHandler Registration Process will exit with error.  To Disable this constraint, set the value to 0.
+    /// </summary>
+    public int RegistrationTimeout { get; set; } = 15000;

```

Comment:
15 seconds seems like a pretty high default?  Are you expecting it to take a that long?

Reply:
no I don't have any expectations.  I'm really open to whatever the team thinks is acceptable.

Reply:
There's no way to have expectations because the user's implementation is impossible to predict.

Reply:
I'm going to resolve this since there doesn't seem to be any issue with the default Timeout length.